node/test_force.js

```diff
@@ -0,0 +1,407 @@
+    log('Generating ' + forcePackageNameForOs(os) + ' package', COLOR.green);
+    var buildNpmPath = os === OS.ios ? path.join(repoDir, 'build', 'build_npm.xml') : path.join(repoDir, 'build_npm.xml');
+    runProcessThrowError('ant -f ' + buildNpmPath);
+    runProcessThrowError('npm install --prefix ' + tmpDir + ' ' + path.join(repoDir, forcePackageNameForOs(os) + '-' + version + '.tgz'));
+}
+
+//
+// Create and compile app 
+//
+function createCompileApp(tmpDir, appType, os) {
+    if (appType === APP_TYPE.native_swift && os === OS.android) return; // that app type doesn't exist for android
+
+    var target = appType + ' app for ' + os;
+    log('==== ' + target + ' ====', COLOR.green);
+    var appName = appType + os + 'App';
+    var appId = '3MVG9Iu66FKeHhINkB1l7xt7kR8czFcCTUhgoA8Ol2Ltf1eYHOU4SqQRSEitYFDUpqRWcoQ2.dBv_a1Dyu5xa';
+    var callbackUri = 'testsfdc:///mobilesdk/detect/oauth/done';
+    var appDir = path.join(tmpDir, appName);
+    var forcePath = path.join(tmpDir, 'node_modules', '.bin', forcePackageNameForOs(os));
+
+    var forceArgs = 'create '
+        + ' --apptype=' + appType
+        + ' --appname=' + appName;
+
+    if (os == OS.ios) {
+        // ios only args
+        forceArgs += ' --companyid=com.mycompany'
+            + ' --organization=MyCompany'
+            + ' --outputdir=' + tmpDir
+            + ' --appid=' + appId
+            + ' --callbackuri=' + callbackUri
+            + ' --startPage=/apex/testPage';
+    }
+    else {
+        // android only args
+        var targetDir;
+
+        if (appType.indexOf('native')>=0) {
+            targetDir = appDir;
+            shelljs.mkdir('-p', targetDir);
+        }
+        else {
+            targetDir = tmpDir;
+        }
+
+        forceArgs += ' --packagename=com.mycompany'
+            + ' --targetdir=' + targetDir
+            + ' --usesmartstore=yes';
+            +' --startpage=/apex/testPage';
+    }
+
+    // Generation
+    var generationSucceeded = runProcessCatchError(forcePath + ' ' + forceArgs, 'GENERATING ' + target);
+
+    if (!generationSucceeded) {
+        return; // no point continuing
+    }
+
+    // Compilation
+    if (appType.indexOf('native') >=0) {
+        if (os == OS.ios) {
+            // IOS - Native
+            editPodfileToUseLocalRepo(appDir);
+            var workspacePath = path.join(appDir, appName + '.xcworkspace');
+            runProcessThrowError('pod update', appDir);    
+            runProcessCatchError('xcodebuild -workspace ' + workspacePath 
+                                 + ' -scheme Pods-' + appName // XXX building Pods-<appName> scheme because <appName> scheme only appears after project is opened in XCode !!
+                                 + ' clean build', 
+                                 'COMPILING ' + target); 
+        }
+        else {
+            // Android - Native
+            runProcessCatchError('./gradlew assembleDebug', 'COMPILING ' + target, appDir);
```

Comment:
@wmathurin I think this line will break on ```Windows```. The equivalent is ```gradlew.bat``` on ```Windows``` if I'm not mistaken. I'm not entirely sure though.